fastlane/lib/fastlane/fast_file.rb

```diff
@@ -355,6 +355,8 @@ def import_from_git(url: nil, branch: 'HEAD', path: 'fastlane/Fastfile', version
             if !version.nil? && current_tag != version
               Actions.sh("cd #{clone_folder.shellescape} && git checkout #{checkout_param.shellescape} #{checkout_path}")
             end
+          else
+            Actions.sh("cd #{clone_folder.shellescape} && git checkout #{checkout_param.shellescape} #{checkout_path}")
           end
```

Comment:
I'd replace this whole `if` with this:

```ruby
          if is_eligible_for_caching
            if version.nil?
              # Update the repo if it's eligible for caching but the version isn't specified
              UI.message("Fetching remote git branches and updating git repo...")
              Helper.with_env_values('GIT_TERMINAL_PROMPT' => '0') do
                Actions.sh("cd #{clone_folder.shellescape} && git fetch --all --quiet && git checkout #{checkout_param.shellescape} #{checkout_path} && git reset --hard && git rebase")
              end
            else
              begin
                # https://stackoverflow.com/a/1593574/865175
                current_tag = Actions.sh("cd #{clone_folder.shellescape} && git describe --exact-match --tags HEAD").strip
              rescue
                current_tag = nil
              end

              if current_tag != version
                Actions.sh("cd #{clone_folder.shellescape} && git checkout #{checkout_param.shellescape} #{checkout_path}")
              end
          else
            Actions.sh("cd #{clone_folder.shellescape} && git checkout #{checkout_param.shellescape} #{checkout_path}")
          end
```

Reply:
Thanks for noticing. Updated statements.